hydroflow/src/builder/surface/pull_chain.rs

```diff
@@ -30,6 +30,24 @@ where
         Self { prev_a, prev_b }
     }
 }
+impl<PrevA, PrevB> TrackPullDependencies for ChainPullSurface<PrevA, PrevB>
+where
+    PrevA: PullSurface + TrackPullDependencies,
+    PrevB: PullSurface<ItemOut = PrevA::ItemOut> + TrackPullDependencies,
+
+    PrevA::InputHandoffs: Extend<PrevB::InputHandoffs>,
+    <PrevA::InputHandoffs as Extend<PrevB::InputHandoffs>>::Extended:
+        PortList<RECV> + PortListSplit<RECV, PrevA::InputHandoffs, Suffix = PrevB::InputHandoffs>,
+{
+    fn insert_dep(&self, e: &mut super::DirectedEdgeSet) -> u16 {
```

Comment:
I think these are how vscode imports them but I think it's more idiomatic to import `DirectedEdgeSet` directly and get rid of the `super::...`